### PR TITLE
go.mod: upgrade golang.org/x/net to v0.7.0 to address CVE-2022-41723

### PR DESCRIPTION
**Pull request by xaxes**
Upgrades golang.org/x/net and some other dependencies due to CVE-2022-41723.

RELEASE NOTES:
- go.mod: upgrade golang.org/x/net to v0.7.0 to address CVE-2022-41723

### CODE REVIEW

File: examples/go.mod [hunk 1]
```diff
@@ -19,9 +19,9 @@ require (
 	github.com/cncf/xds/go v0.0.0-20230105202645-06c439db220b // indirect
 	github.com/envoyproxy/go-control-plane v0.10.3 // indirect
 	github.com/envoyproxy/protoc-gen-validate v0.9.1 // indirect
-	golang.org/x/net v0.5.0 // indirect
-	golang.org/x/sys v0.4.0 // indirect
-	golang.org/x/text v0.6.0 // indirect
+	golang.org/x/net v0.7.0 // indirect
```
Comment (zasweq) [inline examples/go.mod:22]: I see the vulnerability from this, but why are we upgrading the other /sys /text dependencies across the codebase?

**Review by zasweq — CHANGES_REQUESTED** [review]

**Comment by easwars** [comment]
@xaxes : Could you please sign the SLA.

**Comment by easwars** [comment]
@dfawley: The author of this PR has not replied from the time they sent this PR. Should we make a new PR to ensure that these changes are made in time for the 1.54 release?

**Comment by dfawley** [comment]
Yes, we should do this on our own.

**Pull request closed without merging**